src/com/dropbox/core/DbxEntry.java

```diff
@@ -322,6 +335,137 @@ public int hashCode()
+                        JsonReader.nextToken(parser);
+                        if (fieldName.equals("lat_long")) {
+                            location = Location.Reader.read(parser);
+                        } else if (fieldName.equals("time_taken")) {
+                            time_taken = JsonDateReader.Dropbox.readOptional(parser); break;
+                        } else {
+                            JsonReader.skipValue(parser);
+                        }
+                    }
+                    JsonReader.expectObjectEnd(parser);
+                    return new File.PhotoInfo(time_taken, location);
+                }
+            };
+        }
+
+        /**
+         * Holds metadata for videos
+         */
+        public static class VideoInfo {
+            /**
+             * The date and time the video has been shot
+             */
+            public final Date timeTaken;
+
+            /**
+             * The location the video has been shot
+             */
+            public final Location location;
+
+            /**
+             * The duration of the video
+             */
+            public final long duration;
+
+            public VideoInfo(Date timeTaken, Location location, long duration) {
+                this.timeTaken = timeTaken;
+                this.location = location;
+                this.duration = duration;
+            }
+
+            public static JsonReader<VideoInfo> Reader = new JsonReader<VideoInfo>() {
+                @Override
+                public VideoInfo read(JsonParser parser) throws IOException, JsonReadException {
+                    JsonReader.expectObjectStart(parser);
+                    File.Location location = null;
+                    Date time_taken = null;
+                    long duration = 0;
+                    while (parser.getCurrentToken() == JsonToken.FIELD_NAME) {
+                        String fieldName = parser.getCurrentName();
+                        JsonReader.nextToken(parser);
+                        if (fieldName.equals("lat_long")) {
+                            location = Location.Reader.read(parser);
+                        } else if (fieldName.equals("time_taken")) {
+                            time_taken = JsonDateReader.Dropbox.readOptional(parser);
+                        } else if (fieldName.equals("duration")) {
+                            duration = JsonReader.readUnsignedLong(parser);
+                        } else {
+                            JsonReader.skipValue(parser);
+                        }
+                    }
+                    JsonReader.expectObjectEnd(parser);
+                    return new File.VideoInfo(time_taken, location, duration);
+                }
+            };
+        }
+
+        /**
+         * A geo location
+         */
+        public static class Location {
+            /**
+             * The latitude of the location
+             */
+            public final BigDecimal latitude;
```

Comment:
No particular reason. I do store gps coordinates as BigDecimal. I would be
ok though to get them as doubles.

On 31 March 2015 at 21:35, Kannan Goundan <notifications@github.com> wrote:

> In src/com/dropbox/core/DbxEntry.java
> <https://github.com/dropbox/dropbox-sdk-java/pull/21#discussion_r27514623>
> :
>
> > +                        }
> > +                    }
> > +                    JsonReader.expectObjectEnd(parser);
> > +                    return new File.VideoInfo(time_taken, location, duration);
> > +                }
> > +            };
> > +        }
> > +
> > +        /**
> > +         * A geo location
> > +         */
> > +        public static class Location {
> > +            /**
> > +             * The latitude of the location
> > +             */
> > +            public final BigDecimal latitude;
>
> Is there a reason you used BigDecimal and not just double? Were you
> seeing issues when parsing it as a double?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dropbox/dropbox-sdk-java/pull/21/files#r27514623>.
>